src/common/components/proposal-votes/index.tsx

```diff
@@ -29,42 +30,100 @@ interface Voter {
   totalHp: number;
 }
 
+type SortOption = "reputation" | "hp";
+
 interface Props {
   history: History;
   global: Global;
   dynamicProps: DynamicProps;
   proposal: Proposal;
+  votes: string;
+  searchText: string;
   addAccount: (data: Account) => void;
   onHide: () => void;
 }
 
 interface State {
+  page: number;
   loading: boolean;
   voters: Voter[];
+  originalVoters: Voter[];
+  sort: SortOption;
+  noOfPages: number;
+  voter: string;
+  lastDataLength: number;
+  limit: number;
 }
 
 export class ProposalVotesDetail extends BaseComponent<Props, State> {
   state: State = {
-    loading: true,
-    voters: []
+    page: 1,
+    loading: false,
+    voters: [],
+    originalVoters: [],
+    sort: "hp",
+    noOfPages: 0,
+    voter: "",
+    lastDataLength: 0,
+    limit: 1000
   };
 
   componentDidMount() {
     this.load();
   }
 
+  componentDidUpdate(prevProps: Props, prevState: State) {
+    const { voters } = this.state;
+    if (prevProps.searchText !== this.props.searchText) {
+      this.search();
+    }
+    if (prevState.voters !== voters) {
+      this.setState({ noOfPages: Math.ceil(voters.length / 12) });
+    }
+  }
+
+  search = () => {
+    const { originalVoters } = this.state;
+    if (this.props.searchText) {
+      this.setState(
+        {
+          voters: this.state.originalVoters.filter((item) =>
+            item.name.toLocaleLowerCase().includes(this.props.searchText.toLocaleLowerCase())
+          ),
+          page: 1
```

Comment:
Should it be `this.state.page` instead of 1?

Reply:
No, it should be 1. Because we want to show the result of search on the 1st page.